eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -181,12 +181,14 @@ class Setup(datadir: File,
         case true =>
           val host = config.getString("electrum.host")
           val port = config.getInt("electrum.port")
+          val address = InetSocketAddress.createUnresolved(host, port)
           val ssl = config.getString("electrum.ssl") match {
-            case "off" => SSL.OFF
-            case "loose" => SSL.LOOSE
-            case _ => SSL.STRICT // strict mode is the default when we specify a custom electrum server, we don't want to be MITMed
+              case _ if address.getHostName.endsWith(".onion") => SSL.OFF // Tor already adds end-to-end encryption, adding TLS on top doesn't add anything
```

Comment:
Shouldn't this be conditioned on `!"strict"`?

Reply:
What do you mean, prevent ``SSL.OFF`` in case someone explicitly sets strict SSL verification while using an hidden service?

Reply:
Yes, exactly. I think explicit setting should always override anything else.